### PR TITLE
Show mail style correctly

### PR DESCRIPTION
**Pull request by alagane**
Also fixes a bug where a hook was added each time the filter was
called instead of added one time only.

Related to https://github.com/linagora/esn-frontend-inbox/issues/353

### CODE REVIEW

**Comment by alagane** [comment]
See (compared to the example mail in the issue)
image

**Review by fabienmoyon — DISMISSED** [review]

File: src/frontend/js/modules/core.spec.js [hunk 1]
```diff
@@ -420,4 +422,64 @@ describe('The Angular core module', function() {
...
+    });
+
+    it('should prefix ids and class, add div container and modify css selectors', function() {
+      const input =
+      `
+      <style>
+      .test-class {
+        color: #ffffff;
+      }
+      #test-id {
```
Comment (fabienmoyon) [inline src/frontend/js/modules/core.spec.js:444]: can you also add more complex classes in your test like `div#test-id`, `#test-id div.class` , `#test-id > .class` something like that

**Comment by asmaNode** [comment]
Just a remark , maybe it is more ergonomic if we center all text because it is an alert message like this one to attract attention

maintenanceMsg

File: src/frontend/js/modules/core.spec.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+const DOMPurify = require('dompurify/dist/purify.js');
```
Comment (rezk2ll) [inline src/frontend/js/modules/core.spec.js:3]:
```suggestion
```

you don't need this

File: src/frontend/js/modules/core.js [hunk 3]
```diff
@@ -223,16 +224,46 @@ const _ = require('lodash');
...
+      DOMPurify.setConfig({
+        WHOLE_DOCUMENT: true,
+        ADD_ATTR: ['target'],
+        FORBID_TAGS: ['input', 'form']
+      });
+      DOMPurify.addHook('uponSanitizeElement', function(node, data) {
+        if (data.tagName === 'style') {
+          const parsedStyle = parser.parseCSS(node.textContent);
+
+          for (let i = 0; i < parsedStyle.length; i++) {
```
Comment (rezk2ll) [inline src/frontend/js/modules/core.js]:
```suggestion
          parsedStyle = parsedStyle.map(style => {
              style.selector = _computeCssRule(style.selector);
```

Comment (alagane) [reply src/frontend/js/modules/core.js]: The idea is good, though we should not use map in this precise case: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#when_not_to_use_map
I will use map but change the code a little.

Comment (rezk2ll) [reply src/frontend/js/modules/core.js]: you can still use ForEach

File: src/frontend/js/modules/core.spec.js (hunk 1 above)
Comment (alagane) [reply src/frontend/js/modules/core.spec.js:444]: Added successfully :)

**Comment by alagane** [comment]
> Just a remark , maybe it is more ergonomic if we center all text because it is an alert message like this one to attract attention

I am not sure it is related. I just clean a mail style before displaying it, to display it the more accurately possible. I think if the mail should be centered, it is the sender's responsibility.

**Review by rezk2ll — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**